src/Application/Actions/Hooks/StripePaymentsUpdate.php

```diff
@@ -97,21 +97,27 @@ protected function action(Request $request, Response $response, array $args): Re
         $type = $event->type;
         $this->logger->info(sprintf('Received Stripe account event type "%s"', $type));
 
-        switch ($type) {
-            case Event::CHARGE_DISPUTE_CLOSED:
-                return $this->handleChargeDisputeClosed($event, $response);
-            case Event::CHARGE_REFUNDED:
-                return $this->handleChargeRefunded($event, $response);
-            case Event::CHARGE_SUCCEEDED:
-                return $this->handleChargeSucceeded($event, $response);
-            case Event::PAYMENT_INTENT_CANCELED:
-                return $this->handlePaymentIntentCancelled($event, $response);
-            case Event::CUSTOMER_CASH_BALANCE_TRANSACTION_CREATED:
-                return $this->handleCashBalanceUpdate($event, $response);
-            default:
-                $this->logger->warning(sprintf('Unsupported event type "%s"', $type));
+        return match ($type) {
+            Event::CHARGE_DISPUTE_CLOSED =>
+                 $this->handleChargeDisputeClosed($event, $response),
+            Event::CHARGE_REFUNDED =>
+                 $this->handleChargeRefunded($event, $response),
+                // we have to listen for both CHARGE_SUCCEEDED and CHARGE_UPDATED - succeded comes much quicker and
+               // we need to thank the donor for their donation quickly so they know it worked. Updated comes later and
+            // gives us further information that we'll need for charity payouts, e.g. transfer ID.
+            Event::CHARGE_SUCCEEDED =>
+                 $this->handleChargeSucceeded($event, $response),
+            Event::CHARGE_UPDATED =>
+                 $this->handleChargeUpdated($event, $response),
+            Event::PAYMENT_INTENT_CANCELED =>
+                 $this->handlePaymentIntentCancelled($event, $response),
+            Event::CUSTOMER_CASH_BALANCE_TRANSACTION_CREATED =>
+                 $this->handleCashBalanceUpdate($event, $response),
+            default => (function () use ($type, $response) {
+                $this->logger->info(sprintf('Unsupported event type "%s"', $type));
```

Comment:
We're not attempting to handle every event type, so I think it makes sense to downgrade the warning here to an info if not a debug or to nothing at all.

Reply:
I think that as we have none now, it might be helpful to see warnings if we ever add more webhooks but then don't use them. Not a blocker though.

Reply:
Ah ok. I was seeing warnings in my local, maybe we have warnings in local dev envs but not in other environments. I'll put it back to warning in that case, and might look at stopping the local warnings separately.

Reply:
back to warning in 1f63a1a280435071a90a056c24616c7f709590a5